tests/std/tests/P0896R4_ranges_alg_move_backward/test.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <span>
+#include <type_traits>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+struct int_wrapper {
+    int val                 = 10;
+    constexpr int_wrapper() = default;
+    constexpr int_wrapper(int x) : val{x} {}
+    constexpr int_wrapper(int_wrapper&& that) : val{exchange(that.val, -1)} {}
+    constexpr int_wrapper& operator=(int_wrapper&& that) {
+        val = exchange(that.val, -1);
+        return *this;
+    }
+    auto operator<=>(const int_wrapper&) const = default;
+};
+
+// Validate that move_backward_result aliases in_out_result
+STATIC_ASSERT(same_as<ranges::move_backward_result<int, double>, ranges::in_out_result<int, double>>);
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::move_backward(borrowed<false>{}, nullptr_to<int>)),
+    ranges::move_backward_result<ranges::dangling, int*>>);
+STATIC_ASSERT(same_as<decltype(ranges::move_backward(borrowed<true>{}, nullptr_to<int>)),
+    ranges::move_backward_result<int*, int*>>);
+
+static constexpr int expected_output[]      = {13, 42, 1729};

```

Comment:
The other tests have these as static members of the instantiator class. Any reason why to move them out?

Reply:
No reason at all: this is detritus from an experiment.